Components/Sensors/BNO055.cs

```diff
@@ -0,0 +1,333 @@
+        /// </summary>
+        private enum Powermode
+        {
+            POWER_MODE_NORMAL = 0X00,
+            POWER_MODE_LOWPOWER = 0X01,
+            POWER_MODE_SUSPEND = 0X02
+        }
+
+        /// <summary>
+        /// The operation mode of the BNO055. 
+        /// </summary>
+        public enum OperationMode
+        {
+            /* Operation mode settings*/
+            OPERATION_MODE_CONFIG = 0X00,
+            OPERATION_MODE_ACCONLY = 0X01,
+            OPERATION_MODE_MAGONLY = 0X02,
+            OPERATION_MODE_GYRONLY = 0X03,
+            OPERATION_MODE_ACCMAG = 0X04,
+            OPERATION_MODE_ACCGYRO = 0X05,
+            OPERATION_MODE_MAGGYRO = 0X06,
+            OPERATION_MODE_AMG = 0X07,
+            OPERATION_MODE_IMUPLUS = 0X08,
+            OPERATION_MODE_COMPASS = 0X09,
+            OPERATION_MODE_M4G = 0X0A,
+            OPERATION_MODE_NDOF_FMC_OFF = 0X0B,
+            OPERATION_MODE_NDOF = 0X0C
+        }
+
+        /// <summary>
+        /// The possible types of vectors. 
+        /// </summary>
+        public enum VectorType
+        {
+            VECTOR_ACCELEROMETER = Register.BNO055_ACCEL_DATA_X_LSB_ADDR,
+            VECTOR_MAGNETOMETER = Register.BNO055_MAG_DATA_X_LSB_ADDR,
+            VECTOR_GYROSCOPE = Register.BNO055_GYRO_DATA_X_LSB_ADDR,
+            VECTOR_EULER = Register.BNO055_EULER_H_LSB_ADDR,
+            VECTOR_LINEARACCEL = Register.BNO055_LINEAR_ACCEL_DATA_X_LSB_ADDR,
+            VECTOR_GRAVITY = Register.BNO055_GRAVITY_DATA_X_LSB_ADDR
+        }
+
+        int ID;
+        byte Address;
+        OperationMode Mode;
+        II2CBus I2C;
+
+        float X, Y, Z;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:UseRobot.BNO055"/> class, which will communicate via
+        /// the given I2C bus. 
+        /// </summary>
+        /// <param name="I2C"> The I2C bus to communicate over. </param>
+        /// <param name="ID"> ID of BNO055. -1 by default. You probably shouldn't change this. </param>
+        /// <param name="Address"> The I2C address of the BNO055. Defaults to 0x28. You probably shouldn't change this. </param>
+        public BNO055(II2CBus I2C, int ID = -1, byte Address = BNO055_ADDRESS_A)
+        {
+            this.ID = ID;
+            this.Address = Address;
+            this.I2C = I2C;
+            Begin();
+        }
+
+        /// <summary>
+        /// Begins the operation of the BNO055 with the target operation mode. NDOF is the default operation mode.
+        /// This method is called from the constructor. Returns whether or not it succeeded. 
+        /// </summary>
+        /// <returns> Whether the BNO055 successfully initialized. </returns>
+        /// <param name="mode"> Operation mode at startup. NDOF by default. </param>
+        public bool Begin(OperationMode mode = OperationMode.OPERATION_MODE_NDOF)
+        {
+            try { Write8((byte)Register.BNO055_PAGE_ID_ADDR, 0); }
+            catch { goto GoAgain; }
```

Comment:
This could infinitely loop if there is no device connected, which is no good. Maybe add an attempt count limit?